### PR TITLE
Support jetty-eeX-env.xml, as per jetty-eeX-web.xml

### PR DESCRIPTION
**Pull request by janbartel**
Need to support ee specific jetty-env.xml file, just as we do for jetty-web.xml

### CODE REVIEW

File: jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java [hunk 1]
```diff
@@ -47,6 +47,8 @@ public class EnvConfiguration extends AbstractConfiguration
     private static final Logger LOG = LoggerFactory.getLogger(EnvConfiguration.class);
 
     private static final String JETTY_ENV_BINDINGS = "org.eclipse.jetty.jndi.EnvConfiguration";
```
Comment (joakime) [inline jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java:49]: this binding key isn't ee10 specific, should this be in core?

File: jetty-ee11/jetty-ee11-plus/src/main/java/org/eclipse/jetty/ee11/plus/webapp/EnvConfiguration.java [hunk 2]
```diff
@@ -47,6 +47,8 @@ public class EnvConfiguration extends AbstractConfiguration
     private static final Logger LOG = LoggerFactory.getLogger(EnvConfiguration.class);
 
     private static final String JETTY_ENV_BINDINGS = "org.eclipse.jetty.jndi.EnvConfiguration";
```
Comment (joakime) [inline jetty-ee11/jetty-ee11-plus/src/main/java/org/eclipse/jetty/ee11/plus/webapp/EnvConfiguration.java:49]: this name isn't ee11 specific, shouldn't this be in jetty-core?

File: jetty-ee9/jetty-ee9-plus/src/main/java/org/eclipse/jetty/ee9/plus/webapp/EnvConfiguration.java [hunk 3]
```diff
@@ -50,6 +49,8 @@ public class EnvConfiguration extends AbstractConfiguration
     private static final Logger LOG = LoggerFactory.getLogger(EnvConfiguration.class);
 
     private static final String JETTY_ENV_BINDINGS = "org.eclipse.jetty.jndi.EnvConfiguration";
```
Comment (joakime) [inline jetty-ee9/jetty-ee9-plus/src/main/java/org/eclipse/jetty/ee9/plus/webapp/EnvConfiguration.java:51]: This name isn't ee9 specific, shouldn't it be in jetty core?

File: jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java (hunk 1 above)
Comment (janbartel) [reply jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java:49]: Maybe but there's no good place to put it (yet?).

File: jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java [hunk 4]
```diff
@@ -241,6 +230,43 @@ protected void createEnvContext(WebAppContext wac)
...
+
+            return null;
+        }
+        catch (Exception e)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Error resolving WEB-INF/" + xmlFile, e);
+            return null;
+        }
+    }
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-plus/src/main/java/org/eclipse/jetty/ee10/plus/webapp/EnvConfiguration.java]: I think the following is clearer and the exception will have the detail needed:

```suggestion
    private Resource resolveJettyEnvXml(Resource webInf)
    {
        try
        {
            if (webInf == null || !webInf.isDirectory())
                return null;

            //try to find jetty-ee10-env.xml
            Resource xmlResource = webInf.resolve(JETTY_EE10_ENV_XML_FILENAME);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            //failing that, look for jetty-env.xml
            xmlResource = webInf.resolve(JETTY_ENV_XML_FILENAME);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            return null;
        }
        catch (Exception e)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Error resolving", e);
            return null;
        }
    }
```

File: jetty-ee11/jetty-ee11-plus/src/main/java/org/eclipse/jetty/ee11/plus/webapp/EnvConfiguration.java [hunk 5]
```diff
@@ -241,6 +229,43 @@ protected void createEnvContext(WebAppContext wac)
...
+                return xmlResource;
+
+            return null;
+        }
+        catch (Exception e)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Error resolving WEB-INF/" + xmlFile, e);
+            return null;
+        }
```
Comment (gregw) [inline jetty-ee11/jetty-ee11-plus/src/main/java/org/eclipse/jetty/ee11/plus/webapp/EnvConfiguration.java]:
```suggestion
    private Resource resolveJettyEnvXml(Resource webInf)
    {
        try
        {
            if (webInf == null || !webInf.isDirectory())
                return null;

            //try to find jetty-ee10-env.xml
            Resource xmlResource = webInf.resolve(JETTY_EE11_ENV_XML_FILENAME);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            //failing that, look for jetty-env.xml
            xmlResource = webInf.resolve(JETTY_ENV_XML_FILENAME);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            return null;
        }
        catch (Exception e)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Error resolving", e);
            return null;
        }
```

File: jetty-ee9/jetty-ee9-plus/src/main/java/org/eclipse/jetty/ee9/plus/webapp/EnvConfiguration.java [hunk 6]
```diff
@@ -257,4 +247,41 @@ protected void createEnvContext(WebAppContext wac)
...
+
+            return null;
+        }
+        catch (Exception e)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Error resolving WEB-INF/" + xmlFile, e);
+            return null;
+        }
+    }
```
Comment (gregw) [inline jetty-ee9/jetty-ee9-plus/src/main/java/org/eclipse/jetty/ee9/plus/webapp/EnvConfiguration.java]:
```suggestion
    private Resource resolveJettyEnvXml(Resource webInf)
    {
        try
        {
            if (webInf == null || !webInf.isDirectory())
                return null;

            //try to find jetty-ee9-env.xml
            Resource xmlResource = webInf.resolve(JETTY_EE9_ENV_XML);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            //failing that, look for jetty-env.xml
            xmlResource = webInf.resolve(JETTY_ENV_XML);
            if (!Resources.missing(xmlResource))
                return xmlResource;

            return null;
        }
        catch (Exception e)
        {
            if (LOG.isDebugEnabled())
                LOG.debug("Error resolving", e);
            return null;
        }
    }
```

**Review by gregw — CHANGES_REQUESTED** [review]
Minor nit... I found the xmlFile field confusing and was looking for more behaviour for it.   I don't think we need it in the debug as the exception itself will be clear

**Review by gregw — APPROVED** [review]

**Comment by gregw** [comment]
I think this and the jetty-web.xml change should be backported to 12.0

**Pull request merged**